server/test/services/ProgramBlockValidationTest.java

```diff
@@ -0,0 +1,97 @@
+package services;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+import repository.ResetPostgres;
+import services.ProgramBlockValidation.AddQuestionResult;
+import services.program.ProgramDefinition;
+import services.question.types.QuestionDefinition;
+import support.ProgramBuilder;
+
+public class ProgramBlockValidationTest extends ResetPostgres {
+
+  @Test
+  public void canAddQuestion_eligible() throws Exception {
+    QuestionDefinition question = testQuestionBank.applicantName().getQuestionDefinition();
+    ProgramDefinition program =
+        ProgramBuilder.newDraftProgram("program1").withBlock().buildDefinition();
+    assertThat(
+            ProgramBlockValidation.canAddQuestion(
+                program, program.getLastBlockDefinition(), question))
+        .isEqualTo(AddQuestionResult.ELIGIBLE);
+  }
+
+  @Test
+  public void canAddQuestion_duplicate() throws Exception {
+    QuestionDefinition question = testQuestionBank.applicantName().getQuestionDefinition();
+    ProgramDefinition program =
+        ProgramBuilder.newDraftProgram("program1")
+            .withBlock()
+            .withRequiredQuestionDefinition(question)
+            .buildDefinition();
+    assertThat(
+            ProgramBlockValidation.canAddQuestion(
+                program, program.getLastBlockDefinition(), question))
+        .isEqualTo(AddQuestionResult.DUPLICATE);
+  }
+
+  @Test
+  public void canAddQuestion_blockIsSingleQuestion() throws Exception {
+    QuestionDefinition nameQuestion = testQuestionBank.applicantName().getQuestionDefinition();
+    QuestionDefinition fileQuestion = testQuestionBank.applicantFile().getQuestionDefinition();
+    ProgramDefinition program =
+        ProgramBuilder.newDraftProgram("program1")
+            .withBlock()
+            .withRequiredQuestionDefinition(fileQuestion)
+            .buildDefinition();
+    assertThat(
+            ProgramBlockValidation.canAddQuestion(
+                program, program.getLastBlockDefinition(), nameQuestion))
+        .isEqualTo(AddQuestionResult.BLOCK_IS_SINGLE_QUESTION);
+  }
+
+  @Test
+  public void canAddQuestion_cantAddSingleQuestionBlock() throws Exception {
+    QuestionDefinition nameQuestion = testQuestionBank.applicantName().getQuestionDefinition();
+    QuestionDefinition fileQuestion = testQuestionBank.applicantFile().getQuestionDefinition();
+    ProgramDefinition program =
+        ProgramBuilder.newDraftProgram("program1")
+            .withBlock()
+            .withRequiredQuestionDefinition(nameQuestion)
+            .buildDefinition();
+    assertThat(
+            ProgramBlockValidation.canAddQuestion(
+                program, program.getLastBlockDefinition(), fileQuestion))
+        .isEqualTo(AddQuestionResult.CANT_ADD_SINGLE_BLOCK_QUESTION_TO_NON_EMPTY_BLOCK);
+  }
+
+  @Test
+  public void canAddQuestion_cantAddEmuratorQuestionToNonEnumeratorBlock() throws Exception {
```

Comment:
Please add a positive assertion test for enumerator question added to enumerator block